kv/dist_sender.go

```diff
@@ -503,21 +503,24 @@ func (ds *DistSender) sendAttempt(desc *proto.RangeDescriptor, call client.Call)
 		}
 	}
 
-	rpcErr := ds.sendRPC(desc.RaftID, replicas, order, args, reply)
-
-	// For an RPC error to occur, we must've been unable to contact any
-	// replicas. In this case, likely all nodes are down (or not getting back
-	// to us within a reasonable amount of time).
-	// We may simply not be trying to talk to the up-to-date replicas, so
-	// clearing the descriptor here should be a good idea.
-	// TODO(tschottdorf): If a replica group goes dead, this will cause clients
-	// to put high read pressure on the first range, so there should be some
-	// rate limiting here.
-	if rpcErr != nil {
+	var err error
+	if rpcErr := ds.sendRPC(desc.RaftID, replicas, order, args, reply); rpcErr != nil {
```

Comment:
`rpcErr` is no longer needed as a separate variable if it is only used within this if. I'd spell this

    err := ds.sendRPC(...)
    if err != nil {
        // comments...
        ds.rangeCache.EvictCachedRangeDescriptor(...)
    } else {
        err = reply.Header().GoError()
    }